test/Microsoft.AspNetCore.Mvc.Core.Test/Authorization/AuthorizeFilterTest.cs

```diff
@@ -26,6 +26,21 @@ public void InvalidUser()
             Assert.Contains(authorizationContext.HttpContext.User.Identities, i => i.IsAuthenticated);
         }
 
+        [Fact]
+        public async Task AuthorizeFilter_CreatedWithDefaultConstructor_ThrowsWhenOnAuthorizationAsyncIsCalled()
```

Comment:
Can we make a more meaningful test than this? It's not clear why this is an error case.